staging/src/k8s.io/kubectl/pkg/cmd/apply/apply_test.go

```diff
@@ -2269,3 +2261,299 @@ func TestLoadObjects(t *testing.T) {
+		NegotiatedSerializer: resource.UnstructuredPlusDefaultContentConfig().NegotiatedSerializer,
+		Client: fake.CreateHTTPClient(func(req *http.Request) (*http.Response, error) {
+			method := req.Method
+
+			tokens := strings.Split(strings.TrimPrefix(req.URL.Path, "/"), "/")
+
+			if len(tokens) == 2 && tokens[0] == "namespaces" && method == "GET" {
+				name := tokens[1]
+				gvr := schema.GroupVersionResource{Version: "v1", Resource: "namespaces"}
+				ns, err := fakeDynamicClient.Tracker().Get(gvr, "", name)
+				if err != nil {
+					if apierrors.IsNotFound(err) {
+						return &http.Response{StatusCode: http.StatusNotFound, Header: cmdtesting.DefaultHeader()}, nil
+					}
+					t.Fatalf("error getting object: %v", err)
+				}
+				return &http.Response{StatusCode: http.StatusOK, Header: cmdtesting.DefaultHeader(), Body: cmdtesting.ObjBody(codec, ns)}, nil
+			}
+
+			if len(tokens) == 1 && tokens[0] == "namespaces" && method == "POST" {
+				data, err := io.ReadAll(req.Body)
+				if err != nil {
+					t.Fatalf("unexpected error: %v", err)
+				}
+
+				u := &unstructured.Unstructured{}
+				if err := runtime.DecodeInto(codec, data, u); err != nil {
+					t.Fatalf("unexpected error: %v", err)
+				}
+
+				name := u.GetName()
+				ns := u.GetNamespace()
+				gvr := schema.GroupVersionResource{Version: "v1", Resource: "namespaces"}
+
+				existing, err := fakeDynamicClient.Tracker().Get(gvr, ns, name)
+				if err != nil {
+					if apierrors.IsNotFound(err) {
+						existing = nil
+					} else {
+						t.Fatalf("error fetching object: %v", err)
+					}
+				}
+
+				uid := types.UID(fmt.Sprintf("%v", time.Now().UnixNano()))
+				u.SetUID(uid)
+
+				if existing == nil {
+					if err := fakeDynamicClient.Tracker().Create(gvr, u, ns); err != nil {
+						t.Fatalf("error creating object: %v", err)
+					}
+				} else {
+					if err := fakeDynamicClient.Tracker().Update(gvr, u, ns); err != nil {
+						t.Fatalf("error updating object: %v", err)
+					}
+				}
+
+				body := cmdtesting.ObjBody(codec, u)
+
+				return &http.Response{StatusCode: http.StatusOK, Header: cmdtesting.DefaultHeader(), Body: body}, nil
+			}
+
+			t.Fatalf("unexpected request: %v %v\n%#v", req.Method, req.URL, req)
+			return nil, nil
+		}),
+	}
+
+	testdirs := []string{"testdata/prune/simple"}
+	for _, testdir := range testdirs {
+		t.Run(testdir, func(t *testing.T) {
+			cmdtesting.WithAlphaEnvs([]cmdutil.FeatureGate{cmdutil.ApplySet}, t, func(t *testing.T) {
+				manifests := []string{"manifest1", "manifest2"}
+				for i, manifest := range manifests {
+					if i != 0 {
+						t.Logf("will inject failures into future delete operations")
+						failDelete = true
+					}
+					t.Logf("applying manifest %v", manifest)
+
+					var unifiedOutput bytes.Buffer
+					ioStreams := genericclioptions.IOStreams{
+						ErrOut: &unifiedOutput,
+						Out:    &unifiedOutput,
+						In:     bytes.NewBufferString(""),
+					}
+					cmdutil.BehaviorOnFatal(fatalNoExit(ioStreams))
```

Comment:
I notice that there's a helper called `cmdtesting.InitTestErrorHandler()` that catches the errors to fail the test gracefully. Should we inject the testing.T like that to only skip the error we expect (or even require it?)?

Reply:
So I think it's unnecessary to do so.  We simply capture stdout/stderr and verify it using against the expected output.